pkg/services/sso/redhatsso_service.go

```diff
@@ -148,13 +147,17 @@ func (r *redhatssoService) ListServiceAcc(accessToken string, ctx context.Contex
 }
 
 func (r *redhatssoService) RegisterKasFleetshardOperatorServiceAccount(accessToken string, agentClusterId string) (*api.ServiceAccount, *errors.ServiceError) {
+	return r.registerAgentServiceAccount(accessToken, agentClusterId)
+}
+
+func (r *redhatssoService) registerAgentServiceAccount(accessToken string, agentClusterId string) (*api.ServiceAccount, *errors.ServiceError) {
 	svcData, err := r.client.CreateServiceAccount(accessToken, agentClusterId, fmt.Sprintf("service account for agent on cluster %s", agentClusterId))
 	if err != nil {
-		return nil, errors.NewWithCause(errors.ErrorGeneral, err, "failed to update agent service account")
+		return nil, errors.NewWithCause(errors.ErrorGeneral, err, "failed to create agent service account")
 	}
-
 	return convertServiceAccountDataToAPIServiceAccount(&svcData), nil
 }
```

Comment:
This method is called by both `RegisterKasFleetshardOperatorServiceAccount` and `RegisterConnectorFleetshardOperatorServiceAccount` but those two methods doesn't do anything else than calling it and returning its return value.

I think it would be better to:

1. put the code of `registerAgentServiceAccount` into `RegisterKasFleetshardOperatorServiceAccount` 
2. make `RegisterConnectorFleetshardOperatorServiceAccount` call `RegisterKasFleetshardOperatorServiceAccount`
3. delete `registerAgentServiceAccount`

I see two main benefits in doing that:

1. By reading the code it will be easy to understand they both do the same thing
2. We will avoid an useless call in half of the cases

Reply:
For now, I will keep it. because we need to do some rework here once we test it with sso.rh.